app/src/main/java/net/bonysoft/magicmirror/DeviceInformation.java

```diff
@@ -0,0 +1,21 @@
+package net.bonysoft.magicmirror;
+
+import android.os.Build;
+
+public class DeviceInformation {
+
+    /**
+     * Taken from http://stackoverflow.com/a/21505193/1315110
+     */
+    public boolean isEmulator() {
+        return Build.FINGERPRINT.startsWith("generic")
+                || Build.FINGERPRINT.startsWith("unknown")
+                || Build.MODEL.contains("google_sdk")
+                || Build.MODEL.contains("Emulator")
+                || Build.MODEL.contains("Android SDK built for x86")
+                || Build.MANUFACTURER.contains("Genymotion")
+                || (Build.BRAND.startsWith("generic") && Build.DEVICE.startsWith("generic"))
+                || "google_sdk".equals(Build.PRODUCT);
+    }
```

Comment:
Yeah, in other occasions we used just `"generic".equalsIgnoreCase(Build.BRAND)`.
This for sure will cover more cases